File: java/src/main/java/ai/buble/sdk/files/UploadedFile.java
package ai.buble.sdk.files;

/**
 * Uploaded source asset returned by {@code POST /api/v1/files}.
 */
public class UploadedFile {
    private String object;
    private String provider;
    private String url;
    private String key;
    private String fileType;
    private String contentType;
    private long size;
    private String filename;

    public String getObject() {
        return object;
    }

    public void setObject(String object) {
        this.object = object;
    }

    public String getProvider() {
        return provider;
    }

    public void setProvider(String provider) {
        this.provider = provider;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public String getFileType() {
        return fileType;
    }

    public void setFileType(String fileType) {
        this.fileType = fileType;
    }

    public String getContentType() {
        return contentType;
    }

    public void setContentType(String contentType) {
        this.contentType = contentType;
    }

    public long getSize() {
        return size;
    }

    public void setSize(long size) {
        this.size = size;
    }

    public String getFilename() {
        return filename;
    }

    public void setFilename(String filename) {
        this.filename = filename;
    }
}

File: dotnet/src/Buble.Sdk/Files/UploadedFile.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace Buble.Sdk.Files;

public sealed class UploadedFile
{
    [JsonPropertyName("id")]
    public string? Id { get; set; }

    [JsonPropertyName("url")]
    public string? Url { get; set; }

    [JsonPropertyName("file_type")]
    public string? FileType { get; set; }

    [JsonExtensionData]
    public Dictionary<string, JsonElement>? AdditionalProperties { get; set; }
}
